src/app/shared/model/feeder-feed.model.ts

```diff
@@ -0,0 +1,240 @@
+    return 'Slug is reserved';
+  } else if (value.match(/^[0-9a-zA-Z_-]+$/)) {
+    return null;
+  } else {
+    return 'Slug must contain only letters, numbers, underscores and dashes';
+  }
+};
+
+const FEED_FILE_NAME: BaseInvalid = (key: string, value: any): string => {
+  if (value.match(/^[0-9a-zA-Z_.-]+$/)) {
+    return null;
+  } else {
+    return 'File name must contain only letters, numbers, underscores and dashes';
+  }
+};
+
+const TOKENS_JSON: BaseInvalid = (key: string, value: any): string => {
+  const tokens = JSON.parse(value);
+  if (tokens.every((t) => t.label) && tokens.every((t) => t.token)) {
+    return null;
+  } else {
+    return 'Tokens cannot have blank fields';
+  }
+};
+
+const POSITIVE_INT_OR_BLANK: BaseInvalid = (_key: string, value: any): string => {
+  if (value && !value.match(/^[1-9][0-9]*$/)) {
+    return 'Enter a positive number';
+  }
+  return null;
+};
+
+export class FeederFeedModel extends BaseModel {
+  id: number;
+  tokens = [];
+
+  SETABLE = [
+    'title',
+    'slug',
+    'fileName',
+    'private',
+    'tokensJson',
+    'url',
+    'newFeedUrl',
+    'enclosurePrefix',
+    'displayEpisodesCount',
+    'displayFullEpisodesCount',
+    'billboardAds',
+    'houseAds',
+    'paidAds',
+    'sonicAds'
+  ];
+  title = '';
+  slug = '';
+  fileName = 'feed-rss.xml';
+  private = false;
+  tokensJson = '[]';
+  url = '';
+  newFeedUrl = '';
+  enclosurePrefix = '';
+  displayEpisodesCount = '';
+  displayFullEpisodesCount = '';
+  billboardAds = true;
+  houseAds = true;
+  paidAds = true;
+  sonicAds = true;
+  // episodeOffsetSeconds = '';
+  // includeTags = '';
+  // audioFormat = '';
+
+  VALIDATORS = {
+    title: [UNLESS_DEFAULT(REQUIRED())],
+    slug: [UNLESS_DEFAULT(REQUIRED()), UNLESS_DEFAULT(FEED_SLUG)],
+    fileName: [REQUIRED(), FEED_FILE_NAME],
+    tokensJson: [TOKENS_JSON],
+    url: [URL('Not a valid URL')],
+    newFeedUrl: [URL('Not a valid URL')],
+    enclosurePrefix: [URL('Not a valid URL')],
+    displayEpisodesCount: [POSITIVE_INT_OR_BLANK],
+    displayFullEpisodesCount: [POSITIVE_INT_OR_BLANK]
+  };
+
+  URLS = ['url', 'newFeedUrl', 'enclosurePrefix'];
+
+  constructor(podcast: HalDoc, feed?: HalDoc, loadRelated = true) {
+    super();
+    this.init(podcast, feed, loadRelated);
+  }
+
+  get isDefault(): boolean {
+    return !!this.id && !this.slug;
+  }
+
+  key() {
+    if (this.doc) {
+      return `prx.feed.${this.doc.id}`;
+    } else {
+      return `prx.feed.new.${this.parent.id}`;
+    }
+  }
+
+  related() {
+    return {};
+  }
+
+  decode() {
+    this.id = this.doc['id'] || null;
+    this.title = this.doc['title'] || '';
+    this.slug = this.doc['slug'] || '';
+    this.fileName = this.doc['fileName'] || 'feed-rss.xml';
+    this.private = this.doc['private'] || false;
+    this.tokensJson = JSON.stringify(this.doc['tokens'] || []);
+    this.tokens = JSON.parse(this.tokensJson);
```

Comment:
This gets a bit tricky.  Since `BaseModel` really can only deal with string fields, we're keeping track of parallel `tokenJson` and `this.tokens` array-of-objects here.

Thus the helpers for `setTokens` / `labelChanged` / `labelInvalid` etc.  A bit more work, but comparing the JSON strings makes many other things work as expected.